superset/db_engine_specs/base.py

```diff
@@ -952,9 +952,11 @@ def select_star(  # pylint: disable=too-many-arguments,too-many-locals
         if (show_cols or latest_partition) and not cols:
             cols = database.get_columns(table_name, schema)
 
-        if show_cols:
-            fields = cls._get_fields(cols)
         quote = engine.dialect.identifier_preparer.quote
+        if show_cols:
+            # Explicitly quote all column names, as BigQuery doesn't quote column
+            # names that are also identifiers (eg, "limit") by default.
+            fields = [text(quote(col["name"])) for col in cols]
```

Comment:
is this going to affect all dbs? Do you think we should create a property on the db engine spec like "force_column_quotes"?

Reply:
Right, it will. It should be safe, and note that we use the same method to quote schemas and tables below.

Reply:
@villebro thoughts on this?

Reply:
I would expect the BQ dialect to automatically quote reserved keywords. I tried this quickly on sqlite:

```
create table bad_colnames("limit" varchar(10), "offset" varchar(10), regular varchar(10));
```

here's what it rendered:
```sql
SELECT "limit",
       "offset",
       regular
FROM main.bad_colnames
LIMIT 100
OFFSET 0
```
So I'd rather we submit a PR on the BQ connector to make sure reserved keywords are quoted correctly.

Reply:
While I agree this is probably quite safe, I think @eschutho's recommendation to add a property for forcing quotes would be a good solution, so as to avoid adding workarounds that aren't necessary for all engines.

Reply:
Right, SQLite does it the right way, BigQuery is the culprit here.

I'll file a ticket upstream and try to make a PR, and incorporate @eschutho's suggestion into this PR so we don't have to wait for a new release upstream.

Thanks, @villebro!

Reply:
Ahh, looks like someone changed the behavior of `_get_fields` in BigQuery: https://github.com/apache/superset/blob/master/superset/db_engine_specs/bigquery.py#L284-L296

Reply:
This goes back 3+ years: https://github.com/apache/superset/pull/5655